### ISSUE TITLE
Regression after Gradle upgrade from 6.8.3 to 7.0.1: test execution fails (due to "Problem locating LDIF file in schema repository" issues)

### ISSUE THREAD

**Issue by dejan2609** [gradle/gradle#17180]
As in title. Issue occurs after Gradle is upgraded from 6.8.3 to 7.0.1 I managed to reproduce it multiple times (for steps see below). Occurs in multiple environments (see details/notes) below.

### Expected Behavior
Test execution with Gradle 7.0.1 should not fail (at least not due to an error mentioned below).

### Current Behavior
Test execution (but with Gradle 6.8.3) runs as expected.

### Context
 - we are trying to migrate from Gradle 6.8.3 to Gradle 7.0.1
 - this is not a post haste (at least from Apache Kafka's POV, that is)

### Steps to Reproduce 
- checkout this PR: https://github.com/apache/kafka/pull/10606
- execute gradle command: `./gradlew clean -PscalaVersion=2.12 test -x :clients:test` (there is a reason why we are skipping one module, see notes below)
- dozen of tests execution fails with error like this: 
```
GssapiAuthenticationTest > testServerNotFoundInKerberosDatabase() FAILED
    org.apache.directory.api.ldap.schema.extractor.UniqueResourceException: Problem locating LDIF file in schema repository
    Multiple copies of resource named 'schema/ou=schema/cn=apachedns/ou=objectclasses/m-oid=1.3.6.1.4.1.18060.0.4.2.3.9.ldif' located on classpath at urls
        jar:file:/home/dejan/.gradle/caches/modules-2/files-2.1/org.apache.directory.api/api-all/1.0.2/5a0f957dfe453ecaa6e7cbb19d2698083116e123/api-all-1.0.2.jar!/schema/ou%3dschema/cn%3dapachedns/ou%3dobjectclasses/m-oid%3d1.3.6.1.4.1.18060.0.4.2.3.9.ldif
        jar:file:/home/dejan/.gradle/caches/modules-2/files-2.1/org.apache.directory.api/api-ldap-schema-data/1.0.0/eca194f6baca671c466a4eb3ebc669856d997baa/api-ldap-schema-data-1.0.0.jar!/schema/ou%3dschema/cn%3dapachedns/ou%3dobjectclasses/m-oid%3d1.3.6.1.4.1.18060.0.4.2.3.9.ldif
        at org.apache.directory.api.ldap.schema.extractor.impl.DefaultSchemaLdifExtractor.getUniqueResource(DefaultSchemaLdifExtractor.java:366)
        at org.apache.directory.api.ldap.schema.extractor.impl.DefaultSchemaLdifExtractor.getUniqueResourceAsStream(DefaultSchemaLdifExtractor.java:343)
        at org.apache.directory.api.ldap.schema.extractor.impl.DefaultSchemaLdifExtractor.extractFromClassLoader(DefaultSchemaLdifExtractor.java:381)
        at org.apache.directory.api.ldap.schema.extractor.impl.DefaultSchemaLdifExtractor.extractOrCopy(DefaultSchemaLdifExtractor.java:169)
        at org.apache.directory.api.ldap.schema.extractor.impl.DefaultSchemaLdifExtractor.extractOrCopy(DefaultSchemaLdifExtractor.java:190)
        at kafka.security.minikdc.MiniKdc.initDirectoryService(MiniKdc.scala:135)
        at kafka.security.minikdc.MiniKdc.start(MiniKdc.scala:121)
        at kafka.api.SaslSetup.initializeKerberos(SaslSetup.scala:85)
        at kafka.api.SaslSetup.initializeKerberos$(SaslSetup.scala:82)
        at kafka.server.GssapiAuthenticationTest.initializeKerberos(GssapiAuthenticationTest.scala:45)
        at kafka.api.SaslSetup.startSasl(SaslSetup.scala:71)
        at kafka.api.SaslSetup.startSasl$(SaslSetup.scala:63)
        at kafka.server.GssapiAuthenticationTest.startSasl(GssapiAuthenticationTest.scala:45)
        at kafka.server.GssapiAuthenticationTest.setUp(GssapiAuthenticationTest.scala:65)
```

### Your Environment
Actually, there are two environments (and both of them show same results, i.e. test that fail): 
- Jenkins build https://ci-builds.apache.org/blue/organizations/jenkins/Kafka%2Fkafka-pr/detail/PR-10606/1/tests/
- my laptop environment: 
```
dejan@dejan-HP-ProBook-450-G7:~$ java -version
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (build 1.8.0_292-8u292-b10-0ubuntu1~20.04-b10)
OpenJDK 64-Bit Server VM (build 25.292-b10, mixed mode)
dejan@dejan-HP-ProBook-450-G7:~$ date
сре, 12. мај 2021.  22:20:54 CEST
dejan@dejan-HP-ProBook-450-G7:~$
```

Notes: 
- I did some research here https://github.com/apache/kafka/pull/10606#issuecomment-840090090 and here are some Maven related links/resources: 
    - https://issues.apache.org/jira/browse/DIRSERVER-1606
    - https://issues.apache.org/jira/browse/DIRSERVER-2066
    - https://stackoverflow.com/questions/17217465/apache-directory-ldap-api-getting-up-and-running
- PR mentioned above also contains shadow plugin version change (6.1.0 -->> 7.0.0)  important note: compilation yields same results (i.e. test compilation fails) with both shadow plugin versions
- I mentioned above that gradle command should be issued like this `./gradlew clean -PscalaVersion=2.12 test -x :clients:test`. Reason for skipping test in a `clients` module is to separate from another batch of test that fail due to a completely separate issue (that occur only is build is executed on top of JDK 1.8.0_292). I will put a link to a comment that describes these (separate) issue here just for a reference: https://github.com/apache/kafka/pull/10606#issuecomment-840081435

**Comment by dejan2609** [comment]
Two more notes after some one-on-one brainstorming: 
 - not sure who should be responsible to solve this: **_Gradle_** or **_Apache Directory server_**
 - it seems that there is a workaround (according to JIRA issues mentioned above it boils to an exclusion)

**Comment by dejan2609** [comment]
After some more digging: 
- project that has issues uses Apache Directory server library version apacheds **_2.0.0-M24_**
- it seems that there is a workaround for this version
- on top of this: apacheds released more recent version **_2.0.0-M24_** (that solve this, as they claim)

@someone at Gradle: please take all of this into account during a triage.

**Comment by dejan2609** [comment]
Closed in favor of #17187
